test/methods.js

```diff
@@ -0,0 +1,300 @@
+                { name: 'getWebhook', stub: 'get', options: { webhookId: 123 }, expectedRequest: {url: "2.0/webhooks/123" }},
+                { name: 'listWebhooks', stub: 'get', options: {}, expectedRequest: {url: "2.0/webhooks/" }},
+                { name: 'resetSharedSecret', stub: 'post', options: { webhookId: 123 }, expectedRequest: {url: "2.0/webhooks/123/resetsharedsecret" }},
+            ]
+        },
+        {
+            name: 'workspaces',
+            methods: [
+                { name: 'listWorkspaces', stub: 'get', options: {}, expectedRequest: {url: "2.0/workspaces/"}},
+                { name: 'getWorkspace', stub: 'get', options: {workspaceId: 123}, expectedRequest: {url: "2.0/workspaces/123"}},
+                { name: 'listWorkspaceFolders', stub: 'get', options: {workspaceId: 123}, expectedRequest: {url: "2.0/workspaces/123/folders"}},
+                { name: 'createWorkspace', stub: 'post', options: {}, expectedRequest: {url: "2.0/workspaces/"}},
+                { name: 'createFolder', stub: 'post', options: {workspaceId: 123}, expectedRequest: {url: "2.0/workspaces/123/folders"}},
+                { name: 'deleteWorkspace', stub: 'delete', options: {workspaceId: 123}, expectedRequest: {url: "2.0/workspaces/123"}},
+                { name: 'updateWorkspace', stub: 'put', options: {workspaceId: 123}, expectedRequest: {url: "2.0/workspaces/123"}},
+                { name: 'copyWorkspace', stub: 'post', options: {workspaceId: 123}, expectedRequest: {url: "2.0/workspaces/123/copy"}},
+            ]
+        },
+    ];
+
+    _.each(testGroups, function (testGroup) {
+        describe('#' + testGroup.name, function () {
+            _.each(testGroup.methods, function (method) {
+                describe('#' + method.name, function () {
+                    var stub;
+
+                    beforeEach(function () {
+                        stub = sinon.stub(httpUtils, method.stub);
+                    });
+
+                    afterEach(function () {
+                        stub.restore();
+                    });
+
+                    it('method exists', function () {
+                        smartsheet.should.have.property(testGroup.name);
+                        smartsheet[testGroup.name].should.have.property(method.name);
+                    });
+
+                    it('calls utils once', function () {
+                        smartsheet[testGroup.name][method.name](method.options);
+                        stub.callCount.should.be.equal(1);
+                    });
+
+                    it('multiple requests are correct', function () {
+                        smartsheet[testGroup.name][method.name](method.options);
+                        smartsheet[testGroup.name][method.name](method.options);
+                        stub.args[0][0].should.have.properties(method.expectedRequest);
+                    });
```

Comment:
This may not catch scenarios where other endpoints interfere with this one, i.e. where this method is stateful.  For example, if a method always sets a shared value to the same thing each time, it may not notice the mutation it's caused in these tests.

Reply:
You're right, it won't. There was a bug earlier where calling the same endpoint twice would invalidate the Url. I couldn't think of an easy, generalized way to do that. Let me know if you have any ideas